app/[lang]/chapters/[slug]/[lesson]/layout.tsx

```diff
@@ -14,6 +15,11 @@ export default function Layout({ children, params }) {
     <div className={`${theme} flex flex-col`}>
       <div className="fix-grow-issue flex min-h-screen flex-col overflow-hidden">
         <Navbar params={params} />
+        <Chip
+          position={1}
+          coins={1.49}
+          img={'/assets/images/chapter-1-holocat.jpg'}
+        />
```

Comment:
Let's leave it up to the Chip's parent to determine what it renders:
```jsx
<Chip img={...}>
  <span>#1</span>
</Chip>
```

Where both `img` and children should be optional.

Reply:
Sounds like a good idea to make the component more generalized.

However, the component, as per our primary use case, needs to shuffle between two states, with different designs for the states.

https://github.com/saving-satoshi/saving-satoshi/assets/85434418/1d1f3089-2b58-4dd0-936f-4b56ebc47dee

So in making the component more generalized, it might become difficult to keep track of which child span of the Chip component should be displayed and with which color and state combination.

Reply:
I understand what you are saying but in that case lets create a separate component for that while the Chip component just does what I mentioned. This separate component would then use the Chip component, so:

```jsx
export default function Chip({ children }: { children?: React.ReactNode }) {
  return <div className="chip-classes">{children}</div>
}

export default function ProfileChip({ ranking, balance, img }: { ranking: number; balance: number; img: string }) { 
  // Do what you are currently doing in your Chip.tsx here.
  return <Chip>
    ...
  </Chip>
}
```

This will allow us to reuse the Chip component for other stuff.

Reply:
Seems like a logical approach to go with.
Let me implement it!